cypress/integration/task_actions.ts

```diff
@@ -0,0 +1,81 @@
+/// <reference types="Cypress" />
+
+describe("Task Action Buttons", function() {
+  before(() => {
+    cy.login();
+  });
+
+  describe("Based on the state of the task, some buttons should be disabled and others should be clickable. Clicking on buttons produces banners messaging if the action succeeded or failed.", () => {
+    beforeEach(() => {
+      cy.preserveCookies();
+    });
+
+    it("Schedule button should be disabled", () => {
+      cy.visit(taskRoute1);
+      cy.dataCy("schedule-task").should("have.css", "pointer-events", "none");
+    });
+
+    it("Clicking Restart button should produce success banner", () => {
+      cy.dataCy("restart-task").click();
+      cy.dataCy(bannerDataCy).contains(restartSuccessBannerText);
+    });
+
+    it("Clicking Unschedule button should produce success banner", () => {
+      cy.dataCy("ellipsis-btn").click();
+      cy.dataCy("unschedule-task").click();
+      cy.dataCy(bannerDataCy).contains(unscheduleSuccessBannerText);
+    });
+
+    it("Abort button should be disabled", () => {
+      cy.dataCy("ellipsis-btn").click();
+      cy.dataCy("abort-task").should("have.css", "pointer-events", "none");
+    });
+
+    it("Clicking on set priority, entering a priority value and submitting should result in a success banner.", () => {
+      cy.dataCy("prioritize-task").click();
+      cy.get(".ant-input-number-input")
+        .clear()
+        .type("99");
+      cy.get(".ant-btn.ant-btn-primary.ant-btn-sm")
+        .contains("Set")
+        .click({ force: true });
+      cy.dataCy(bannerDataCy).contains(prioritySuccessBannerText);
+    });
+
+    it("There should be three visible banners from the previous actions", () => {
+      cy.dataCy(bannerDataCy).should("have.length", 3);
+    });
+
+    it("Clicking on Schedule should produce an error banner", () => {
+      cy.dataCy("schedule-task").click();
+      cy.dataCy(bannerDataCy).contains(scheduleErrorBannerText);
+    });
+
+    it("There should be four banners visible from the previous actions", () => {
+      cy.dataCy(bannerDataCy).should("have.length", 4);
+      cy.dataCy("ellipsis-btn").click();
+    });
+
+    it("Visiting a different task page should clear all banners", () => {
+      cy.visit(taskRoute2);
+      cy.dataCy(bannerDataCy).should("have.length", 0);
+    });
+
+    it("Clicking on Abort should produce a success banner", () => {
```

Comment:
id like to explore how we name these cypress tests in the future. I feel like e2e tests should describe a piece of functionality of the application as a whole. `"Aborting a task should succeed and produce an error banner"`

Reply:
that test name sounds no more descriptive to me than what he has. I want to know how the feature works -- `Successfully aborting a task shows success banner (and changes task status?)`